src/components/link/link.test.tsx

```diff
@@ -0,0 +1,569 @@
+    linkElement.focus();
+    await user.keyboard("{Enter}");
+
+    expect(onKeyDownFn).toHaveBeenCalled();
+  });
+
+  it("should not fire `onClick` when a key is pressed but no `onClick` prop is passed", async () => {
+    const onClickFn = jest.fn();
+    const onKeyDownFn = jest.fn();
+    const user = userEvent.setup();
+    render(<Link onKeyDown={onKeyDownFn} href="#" />);
+
+    const linkElement = screen.getByTestId("link-anchor");
+    linkElement.focus();
+    await user.keyboard("{Enter}");
+
+    expect(onKeyDownFn).toHaveBeenCalled();
+    expect(onClickFn).not.toHaveBeenCalled();
+  });
+});
+
+test("component should render a button element when `onClick` prop is passed", () => {
+  render(<Link onClick={() => {}} />);
+
+  const buttonElement = screen.getByRole("button");
+
+  expect(buttonElement).toBeInTheDocument();
+});
+
+test("when rendered as an `<a>` element, it should set the expected aria attributes", () => {
+  render(<Link aria-label="test" />);
+
+  const linkElement = screen.getByTestId("link-anchor");
+
+  expect(linkElement).toHaveAttribute("aria-label", "test");
+});
+
+test("when rendered as a `button` element, it should set the aria attributes on the button", () => {
+  render(<Link onClick={() => {}} aria-label="test" />);
+
+  const buttonElement = screen.getByRole("button");
+
+  expect(buttonElement).toHaveAttribute("aria-label", "test");
+});
+
+test("when `removeAriaLabelOnIcon` is true, it should set aria-label as undefined on the icon", () => {
+  render(
+    <Link
+      onClick={() => null}
+      icon="home"
+      aria-label="test"
+      removeAriaLabelOnIcon
+    />
+  );
+
+  const iconElement = screen.getByTestId("icon");
+
+  expect(iconElement).not.toHaveAttribute("aria-label");
+});
+
+// Test is just for coverage
+test("neutral `variant` has the expected styling when `isDarkBackground` is false", () => {
+  render(
+    <Link
+      href="foo.com"
+      isDarkBackground={false}
+      icon="home"
+      variant="neutral"
+      data-role="link"
+    />
+  );
+
+  const linkElement = screen.getByTestId("link");
+  const iconElement = screen.getByTestId("icon");
+
+  expect(linkElement).toHaveStyle("color: var(--colorsActionMajorYin090)");
+  expect(iconElement).toHaveStyle("color: var(--colorsActionMajorYin090)");
+});
+
+// Test is just for coverage
+test("neutral `variant` has the expected styling when `isDarkBackground` is false and is hovered over", () => {
+  const user = userEvent.setup();
+  render(
+    <Link
+      href="foo.com"
+      isDarkBackground={false}
+      icon="home"
+      variant="neutral"
+      data-role="link"
+    />
+  );
+
+  const linkElement = screen.getByTestId("link");
+  const iconElement = screen.getByTestId("icon");
+
+  user.hover(linkElement);
+
+  expect(linkElement).toHaveStyle("color: var(--colorsActionMajor600)");
+  expect(iconElement).toHaveStyle("color: var(--colorsActionMajor600)");
+});
```

Comment:
**comment**: I've seen you have done this in previous tests, I would also make this test `async` and use `await` for userEvents as it might cause some unexpected behaviour.  Same for the next few tests.